packages/app/src/AppNav.tsx

```diff
@@ -426,6 +423,64 @@ function PresetSearchLink({ query, name }: { query: string; name: string }) {
   );
 }
 
+function SearchInput({
+  placeholder,
+  value,
+  onChange,
+}: {
+  placeholder: string;
+  value: string;
+  onChange: (arg0: string) => void;
+}) {
+  return (
+    <Input
+      placeholder={placeholder}
+      value={value}
+      onChange={e => onChange(e.currentTarget.value)}
+      icon={<i className="bi bi-search fs-8 ps-1" />}
+      rightSection={
+        value && (
+          <CloseButton size="xs" radius="xl" onClick={() => onChange('')} />
+        )
+      }
+      mt={8}
+      size="xs"
+      variant="filled"
+      radius="xl"
+      sx={{
+        input: {
+          minHeight: 28,
+          height: 28,
+          lineHeight: 28,
+        },
+      }}
+    />
+  );
+}
+
+function useSearchableList<T extends { name: string }>({
+  items,
+}: {
+  items: T[];
+}) {
+  const [q, setQ] = useState('');
+
+  const filteredList = useMemo(() => {
+    if (q === '') {
+      return items;
+    }
+    return items.filter(item =>
```

Comment:
just an fyi, we do have fuse.js already included and can use it in the future as well for fancy search :)